server/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -182,9 +194,56 @@ private void buildResponse(final ClusterStateRequest request,
                 }
             }
         }
-        listener.onResponse(new ClusterStateResponse(currentState.getClusterName(), builder.build(),
-            PublicationTransportHandler.serializeFullClusterState(currentState, Version.CURRENT).length(), false));
+
+        clusterStateSizeByVersionCache.getOrComputeCachedSize(currentState.version(),
+            () -> PublicationTransportHandler.serializeFullClusterState(currentState, Version.CURRENT).length(),
+            ActionListener.wrap(size ->
```

Comment:
NIT: you could use the new `ActionListener#map` here to make this a little nicer :)

Reply:
Today is a day of learning :)